File: app/src/main/java/com/mju/csmoa/util/room/dao/SearchHistoryDao.java
package com.mju.csmoa.util.room.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.mju.csmoa.util.room.entity.SearchHistory;

import java.util.List;

@Dao
public interface SearchHistoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSearchHistory(SearchHistory searchHistory);

    @Delete
    void deleteSearchHistory(SearchHistory searchHistory);

    @Query("SELECT * FROM SearchHistory order by createdAt desc;")
    LiveData<List<SearchHistory>> getSearchHistoryList();

    @Query("DELETE FROM SearchHistory")
    void deleteAllSearchHistory();

}

File: app/src/main/java/com/mju/csmoa/util/room/dao/SearchHistoryDao.kt
package com.mju.csmoa.util.room.dao

import androidx.room.*
import com.mju.csmoa.util.room.entity.SearchHistory
import kotlinx.coroutines.flow.Flow

@Dao
interface SearchHistoryDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSearchHistory(searchHistory: SearchHistory)

    @Delete
    suspend fun deleteSearchHistory(searchHistory: SearchHistory)

//    @Query("SELECT * FROM SearchHistory ORDER BY createdAt DESC;")
//    fun getSearchHistoriesFlow(): Flow<List<SearchHistory>>

    // 리뷰 최근 검색어
    @Query("SELECT * FROM SearchHistory WHERE type = 0 ORDER BY createdAt DESC")
    fun getReviewSearchHistoriesFlow(): Flow<List<SearchHistory>>

    // 레시피 최근 검색어
    @Query("SELECT * FROM SearchHistory WHERE type = 1 ORDER BY createdAt DESC")
    fun getRecipeSearchHistoriesFlow(): Flow<List<SearchHistory>>

//    @Query("DELETE FROM SearchHistory")
//    suspend fun deleteAllSearchHistory()

    // 리뷰 최근 검색어 모두 삭제
    @Query("DELETE FROM SearchHistory WHERE type = 0")
    suspend fun deleteAllReviewSearchHistory()

    // 레시피 최근 검색어 모두 삭제
    @Query("DELETE FROM SearchHistory WHERE type = 1")
    suspend fun deleteAllRecipeSearchHistory()
}